File: on-boarding-samples/on-boarding-sample-java/src/test/java/apio/architect/example/APITests.java
package apio.architect.example;

import khttp.KHttp;
import khttp.responses.Response;
import org.json.JSONArray;
import org.json.JSONObject;
import org.junit.jupiter.api.Test;

import static java.util.Collections.singletonMap;
import static org.junit.jupiter.api.Assertions.assertEquals;

class APITests {

    @Test
    void personEndpointShouldReturnPersonList() {
        Response response = KHttp.get(
            "http://localhost:8080/api/person",
            singletonMap("Accept", "application/json"));

        assertEquals(200, response.getStatusCode());

        JSONObject jsonObject = response.getJsonObject();

        assertEquals(2, jsonObject.get("totalNumberOfItems"));

        JSONArray elements = (JSONArray) jsonObject.get("elements");

        JSONObject alex = (JSONObject) elements.get(0);

        assertEquals("Alex", alex.get("name"));
        assertEquals("Developer", alex.get("jobTitle"));

        JSONObject david = (JSONObject) elements.get(1);

        assertEquals("David", david.get("name"));
        assertEquals("Developer", david.get("jobTitle"));
    }

    @Test
    void personIdEndpointShouldReturnSinglePerson() {
        Response response = KHttp.get(
            "http://localhost:8080/api/person/1",
            singletonMap("Accept", "application/json"));

        assertEquals(200, response.getStatusCode());

        JSONObject jsonObject = response.getJsonObject();

        assertEquals("Alex", jsonObject.get("name"));
        assertEquals("Developer", jsonObject.get("jobTitle"));
    }

}

File: on-boarding-samples/on-boarding-sample-kotlin/src/test/java/apio/architect/example/APITests.kt
package apio.architect.example

import org.json.JSONArray
import org.json.JSONObject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class APITests {

    @Test
    fun `person endpoint should return 🧔🧔`() {
        val response = khttp.get(
                url = "http://localhost:8080/api/person",
                headers = mapOf("Accept" to "application/json"))

        assertEquals(200, response.statusCode)

        assertEquals(2, response.jsonObject["totalNumberOfItems"])

        (response.jsonObject["elements"] as JSONArray).apply {
            with(get(0) as JSONObject) {
                assertEquals("Alex", get("name"))
                assertEquals("Developer", get("jobTitle"))
            }
            with(get(1) as JSONObject) {
                assertEquals("David", get("name"))
                assertEquals("Developer", get("jobTitle"))
            }
        }
    }

    @Test
    fun `person|{id} endpoint should return 🧔`() {
        val response = khttp.get(
                url = "http://localhost:8080/api/person/1",
                headers = mapOf("Accept" to "application/json"))

        assertEquals(200, response.statusCode)

        assertEquals("Alex", response.jsonObject["name"])
        assertEquals("Developer", response.jsonObject["jobTitle"])
    }

}